ocs_ci/ocs/ocs_upgrade.py

```diff
@@ -541,64 +553,112 @@ def run_ocs_upgrade(operation=None, *operation_args, **operation_kwargs):
             upgrade=True
         )
         log.info(f"Disconnected upgrade - new image: {upgrade_ocs.ocs_registry_image}")
+
+    channel = upgrade_ocs.set_upgrade_channel()
     with CephHealthMonitor(ceph_cluster):
-        channel = upgrade_ocs.set_upgrade_channel()
         upgrade_ocs.set_upgrade_images()
-        if upgrade_version != "4.9":
-            # In the case of upgrade to ODF 4.9, the ODF operator should upgrade
-            # OCS automatically.
-            upgrade_ocs.update_subscription(channel)
-        if original_ocs_version == "4.8" and upgrade_version == "4.9":
-            deployment = Deployment()
-            deployment.subscribe_ocs()
-        if (config.ENV_DATA["platform"] == constants.IBMCLOUD_PLATFORM) and not (
-            upgrade_in_current_source
-        ):
-            create_ocs_secret(config.ENV_DATA["cluster_namespace"])
-            for attempt in range(2):
-                # We need to do it twice, because some of the SA are updated
-                # after the first load of OCS pod after upgrade. So we need to
-                # link updated SA again.
-                log.info(
-                    f"Sleep 1 minute before attempt: {attempt+1}/2 "
-                    "of linking secret/SAs"
-                )
-                time.sleep(60)
-                link_all_sa_and_secret_and_delete_pods(
-                    constants.OCS_SECRET, config.ENV_DATA["cluster_namespace"]
+        ui_upgrade_supported = False
+        if config.UPGRADE.get("ui_upgrade"):
+            ocp_version = get_ocp_version()
+            if (
+                Version.coerce(ocp_version) == Version.coerce("4.9")
+                and original_ocs_version == "4.8"
+                and upgrade_version == "4.9"
+            ):
+                ui_upgrade_supported = True
+            else:
+                log.warning(
+                    "UI upgrade combination is not supported. It will fallback to CLI upgrade"
```

Comment:
why we need to fall back to CLI upgrade instead of raising exception? Use is expected to do upgrade through UI

Reply:
@vavuthu Petr suggested this above, and it makes sense to me. What do you think?

Reply:
We have the same fallback also for deployment